app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt

```diff
@@ -5,53 +5,230 @@
 package org.mozilla.fenix.home.sessioncontrol.viewholders.onboarding
 
 import android.view.View
-import androidx.core.view.isVisible
-import androidx.recyclerview.widget.RecyclerView
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Button
+import androidx.compose.material.Card
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.LifecycleOwner
 import org.mozilla.fenix.R
-import org.mozilla.fenix.databinding.NimbusMessageCardBinding
-import org.mozilla.fenix.ext.increaseTapArea
+import org.mozilla.fenix.compose.ComposeViewHolder
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.compose.SectionHeader
 import org.mozilla.fenix.gleanplumb.Message
 import org.mozilla.fenix.home.sessioncontrol.SessionControlInteractor
+import org.mozilla.fenix.nimbus.MessageData
+import org.mozilla.fenix.nimbus.StyleData
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
 
+/**
+ * View holder for the Nimbus Message Card.
+ *
+ * @property interactor [SessionControlInteractor] which will have delegated to all user
+ * interactions.
+ */
 class MessageCardViewHolder(
-    view: View,
-    private val interactor: SessionControlInteractor
-) : RecyclerView.ViewHolder(view) {
+    composeView: ComposeView,
+    viewLifecycleOwner: LifecycleOwner,
+    private val interactor: SessionControlInteractor,
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
+
+    private lateinit var messageGlobal: Message
 
     fun bind(message: Message) {
-        val binding = NimbusMessageCardBinding.bind(itemView)
+        messageGlobal = message
+        interactor.onMessageDisplayed(message)
+    }
```

Comment:
Checking on other viewholders that are already using compose like [RecentlyVisitedViewHolder](https://github.com/mozilla-mobile/fenix/blob/71ac549b8b32466b16d318a1801c4b2b747fd0ce/app/src/main/java/org/mozilla/fenix/home/recentvisits/view/RecentlyVisitedViewHolder.kt#L44), [RecentBookmarksViewHolder](https://github.com/mozilla-mobile/fenix/blob/71ac549b8b32466b16d318a1801c4b2b747fd0ce/app/src/main/java/org/mozilla/fenix/home/recentbookmarks/view/RecentBookmarksViewHolder.kt#L36) and [RecentTabViewHolder]( https://github.com/mozilla-mobile/fenix/blob/71ac549b8b32466b16d318a1801c4b2b747fd0ce/app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabViewHolder.kt#L56), it looks like they are no using the `bind` fun to pass state, it looks like they are observing the state on the store using `observeAsComposableState`, maybe we should follow the same pattern? WDYT?

Reply:
Agreed